### PR TITLE
WIP: Smart contract v2

### PR DESCRIPTION
**Pull request by favadi**

### CODE REVIEW

**Comment by favadi** [comment]
[SmartContract V2 Integration](https://trello.com/c/9DLJotEj/122-smartcontract-v2-integration)

File: blockchain/event_log.go [hunk 1]
```diff
@@ -1,31 +1,129 @@
...
+		// Token-ETH
+		ethAmount.SetInt(tradeLog.DestAmount)
+	} else if tradeLog.EtherReceivalAmount != nil {
+		// Token-Token
+		ethAmount.SetInt(tradeLog.EtherReceivalAmount)
+	}
+
+	// fiat amount = ETH amount * rate
+	ethAmount = ethAmount.Mul(ethAmount, new(big.Float).SetFloat64(rate))
+	ethAmount.Quo(ethAmount, new(big.Float).SetFloat64(math.Pow10(18)))
```
Comment (cyberliem) [inline blockchain/event_log.go]: Should have use math.Pow10(eth.Decimal). Although unlikely that Eth.Decimal will change, yet this practice is recommended for better consistency across the repo

Comment (favadi) [reply blockchain/event_log.go]: Fixed in 86d349e67f62101fa1800781c38a34ba8a1878c2

**Comment by favadi** [comment]
Close in favor of #348

**Pull request closed without merging**